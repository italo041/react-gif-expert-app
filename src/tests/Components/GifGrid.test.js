import { shallow } from "enzyme"
import React from 'react';
import GifGrid from "../../components/GifGrid";
import '@testing-library/jest-dom';
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock('../../hooks/useFetchGifs');

describe('Prueba del componente <GifGrid/>', () => {
    
    const category = 'One punch';
  
    test('Debe de mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: false
        });

        let wrapper = shallow( <GifGrid category = { category }/> );
        
        expect( wrapper ).toMatchSnapshot();
        
    })

    test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquiercosa',
            title: 'Cualquier cosa',
        },
        {
            id: 'ABC',
            url: 'https://localhost/cualquiercosa',
            title: 'Cualquier cosa',
        },
        {
            id: 'ABC',
            url: 'https://localhost/cualquiercosa',
            title: 'Cualquier cosa',
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        let wrapper = shallow( <GifGrid category = { category }/> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe(gifs.length);
   
    })
    
    
})
