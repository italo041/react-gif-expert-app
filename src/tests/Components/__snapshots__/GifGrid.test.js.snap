// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente <GifGrid/> Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One punch
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa"
    />
  </div>
</Fragment>
`;

exports[`Prueba del componente <GifGrid/> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One punch
     
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
